lib/Doctrine/DBAL/Connection.php

```diff
@@ -489,12 +489,14 @@ public function setAutoCommit($autoCommit)
      * Sets the fetch mode.
      *
      * @param integer $fetchMode
-     *
+     * @param mixed $arg2
```

Comment:
Yes, this syntax incompatible with PHP 5.3, but it easy to change brackets [] to array(...). In my case, I use $arg2 - this is column number for fetching.
More examples can find in PHP PDO statement documentation (http://php.net/manual/en/pdostatement.setfetchmode.php):
public bool PDOStatement::setFetchMode ( int $PDO::FETCH_COLUMN , int $colno )
public bool PDOStatement::setFetchMode ( int $PDO::FETCH_INTO , object $object )
public bool PDOStatement::setFetchMode ( int $PDO::FETCH_CLASS , string $classname , array $ctorargs )

$arg2 can be string, object, int. $arg3 only array.